### ISSUE TITLE
Fix `no-duplicate-selectors` false negatives for three or more duplicates

### ISSUE THREAD

**Issue by erosman** [stylelint/stylelint#7583]
### What minimal example or steps are needed to reproduce the bug?

```css
a[href*="abc"],
a[href*=""],
a[href*=""],
a[href*=""] {
  color: #fd0;
}
```

### What minimal configuration is needed to reproduce the bug?

default

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGBtAFgJxgMwFQF4AdZAIzGIF0AaQgO1UxwOMpvvSzyJFboc+Y8QFAATA6IkZAA2EDPBEBibFAAMAbjoBfEFRDYAltJgA5JAFs4iGAA8LAB2MA6MAGdXu8BFqGA5ghBxWkliWwAXGFooV2IFFAlJEWJXMIBPY2kDWjCAWkgfA18clKQopAwoFmCRCm1PfL8AMTlzJDCAgCtXb09Yew9EIJCQFPSYTOzYpJBpNpgU4jZh0Yys3IbC4rDSqHLKhGnZiIWeWh0tIA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.3.0",
  "stylelint-config-standard": "36.0.0"
}
```

### What did you expect to happen?

Error should occur on line 3 & 4 and refer to line 2 e.g.
Line 2 selector is not a duplicate of line 1 selector.

---
PS.  (on another note) Is there a rule for empty selector value? :thinking: 

### What actually happened?

ERROR Unexpected duplicate selector "a[href*=""]", first used at line 1 (no-duplicate-selectors) [2:1-12]

**Comment by ybiquitous** [comment]
@erosman Thanks for the report. As you commented, this is a bug.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
> PS. (on another note) Is there a rule for empty selector value? 🤔

Sadly, we have no such a rule in the [built-in rules](https://stylelint.io/user-guide/rules/). If you're interested, feel free to open a new issue.

**Comment by ybiquitous** [comment]
For your reference, this bug is not only for empty values.

```css
a[href*="abc"], /* this line only is reported */
a[href*="abc"],
a[href*="abc"],
a[href*=""] {
  color: #fd0;
}
```

See the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGBtAFgJxgMwFQF4AdZAIzGIF0AaQgO1UxwOKTMpvvSzyNPJGp0G3ZiEoACYHXHjIAGwgZ44gMTYoABgDcdAL4gqIbAEs5MAHJIAtnEQwAHtYAOZgHRgAzh4PgItEwDmCCBStDLEDgAuMLRQHsTKKNIy4sQekQCeZnLGtJEAtJD+xgH56UixSBhQxMkUej5FgQBiilZIkcEAVh5+PrBO3oih4SDpWTA5eQmpIHIdMOnEHKPj2bkFTSVlkRVQVTUIs-PRS2K0+rpAA).

**Comment by erosman** [comment]
> @erosman Thanks for the report. As you commented, this is a bug.
> 
> I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.
> 
> There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

Is the code for the rule in [/lib/rules/no-duplicate-selectors/index.mjs](https://github.com/stylelint/stylelint/blob/a5965a09b557fd4a5420a6b48f1c1353ecf83aa2/lib/rules/no-duplicate-selectors/index.mjs)?

TBH, my writing style is considerably different for me to contribute to the code.

**Comment by ybiquitous** [comment]
> Is the code for the rule in [/lib/rules/no-duplicate-selectors/index.mjs](https://github.com/stylelint/stylelint/blob/a5965a09b557fd4a5420a6b48f1c1353ecf83aa2/lib/rules/no-duplicate-selectors/index.mjs)?

Yes.

> TBH, my writing style is considerably different for me to contribute to the code.

Regarding coding style, we keep consistency thanks to Prettier and ESLint.

**Comment by erosman** [comment]
BTW, is CLA required by StyleLint?
_(It prevented me from processing a PR in ESLint.)_

**Comment by ybiquitous** [comment]
No, CLaa is not required.
